app/lib/server/functions/saveUser.js

```diff
@@ -171,6 +171,11 @@ export const saveUser = function(userId, userData) {
 
 		const _id = Accounts.createUser(createUser);
 
+		// set userData.name to empty string to guarantee it's not undefined
+		if (!userData.name) {
+			userData.name = '';
```

Comment:
Why not? That way if I as admin edit a user that doesn't have a name `name === undefined` I'll update it to `name === ''` and all of his subscriptions.

Reply:
and how would you clear the username after it's been set to something?